File: csharp/BronKerboschStudy/NumbersGame.cs
using System;
using System.Globalization;

namespace BronKerboschStudy
{
    internal static class NumbersGame
    {
        public static int ParseInt(string numstr)
        {
            int factor = 1;
            if (numstr.EndsWith('k', StringComparison.Ordinal))
            {
                factor = 1_000;
                numstr = numstr[..^1];
            }
            else if (numstr.EndsWith('M', StringComparison.Ordinal))
            {
                factor = 1_000_000;
                numstr = numstr[..^1];
            }
            return int.Parse(numstr, CultureInfo.InvariantCulture) * factor;
        }
    }
}

File: fsharp/BronKerboschStudy/NumbersGame.fs
namespace BronKerboschStudy

module NumbersGame =
    let ParseInt (numstr: string) : int =
        let suffix_length, factor =
            match numstr with
            | s when s.EndsWith('k') -> 1, 1_000
            | s when s.EndsWith('M') -> 1, 1_000_000
            | _ -> 0, 1

        int (numstr[0 .. numstr.Length - 1 - suffix_length]) * factor
